src/libraries/System.Private.CoreLib/src/System/Decimal.DecCalc.cs

```diff
@@ -392,6 +467,30 @@ private static uint Div96By64(ref Buf12 bufNum, ulong den)
                 return quo;
             }
 
+
+            /// <summary>
+            /// Perform multiplication between 64 and 32 bit numbers, returning lower 64 bits in <paramref name="low"/>
+            /// </summary>
+            /// <returns>hi bits of the result</returns>
+            /// <remarks>returns nuint instead of uint to skip clearing upper 32bits on 64bit platforms</remarks>
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            private static nuint BigMul64By32(ulong a, uint b, out ulong low)
```

Comment:
Same general comment. This is another case where this should really just be a general purpose opt in the JIT, recognizing `ulong * uint` and optimizing it accordingly, rather than requiring a full `ulong * ulong`, just opting to do whichever is most efficient.

Reply:
Sounds good if the JIT could recognice it and optimize the code. 
Since the currently most popular approach to the get more than 64bit result is to call BigMul I hope that that method can be optimized in the future.

I created an internal overload of BigMul in the time beeing, which can easily be removed when the JIT can optimize 32*64 bit => 128bit multiplications